goshopify.go

```diff
@@ -414,13 +417,13 @@ func (c *Client) Count(path string, options interface{}) (int, error) {
 // The options argument is used for specifying request options such as search
 // parameters like created_at_min
 // Any data returned from Shopify will be marshalled into resource argument.
-func (c *Client) CreateAndDo(method, path string, data, options, resource interface{}) error {
+func (c *Client) CreateAndDo(method, path string, data, options, resource interface{}, headers *http.Header) error {
```

Comment:
So this is technically considered a "breaking change" since `CreateAndDo` is public. While normal users probably wouldn't be using this method unless they had to work with custom resources or things the current lib doesn't implement it would break some users.

2 options 
1. get the ok the break this since it is a very small break
2. Add a `CreateAndDoWithHeaders()` that has your new parameter, move all the code to that and have `CreateAndDo` take the old params and just return a call to `CreateAndDoWithHeaders()` while passing nil for the `headers` param.

You could make `createAndDoWithHeaders()` private if the lib+ you isn't settled on this exact api yet since making it public would add more things to maintain compatibility with.

I'm just in github ui right now but I'll see if I can pull up the repo later to get a full view of this. Passing around a pointer to headers seems odd and while I agree you need access to the headers it might be better as a return value or something else instead since I'm pretty sure you just need this for extracting headers? Maybe a mix of passing a non pointer to control input + returning the response headers in a return value instead would be best.